test/bpf_tests/bpf_test.go

```diff
@@ -0,0 +1,443 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
```

Comment:
To pick these up as part of the privileged test suite:

```suggestion

//go:build privileged_tests
```

(These should also be run from `make tests-privileged` so that they're automatically hooked into CI, please check that's the case)